### PR TITLE
Add support for multiple ORs

### LINKED ISSUE #307: Cannot Composing schemas more than 2 schemas
**Describe the bug**

When I tried to compose schemas more than 2, then it doesn't support and throw me an error.

**To Reproduce**

```rb
require "dry-validation"

class UserContract < Dry::Validation::Contract 
  IndividualSchema = Dry::Schema.JSON do
    required(:first_name).filled(:str?)
    required(:last_name).filled(:str?)
  end

  CompanySchema = Dry::Schema.JSON do
    required(:name).filled(:str?)
    required(:owner_name).filled(:str?)
  end

  AnonymousSchema = Dry::Schema.JSON do
    required(:country).filled(:str?)
  end

  params do
    required(:details).schema(IndividualSchema | CompanySchema | AnonymousSchema)
  end
end

contract = UserContract.new

p contract.call(details: { first_name: "John", last_name: "Doe" }).success?
```

It raise an exception below:

```
dry-schema-1.5.2/lib/dry/schema/message/or.rb:16:in `map': undefined method `path' for #<Dry::Schema::Message::Or::MultiPath:0x00007fdceb290a7
0> (NoMethodError)
```

**Expected behavior**

I hope we should be able to compose more than 2 schemas.

**Your environment**

* Ruby version: ruby 2.7.1p83
* dry-schema: 1.5.2

### PR DESCRIPTION
**Pull request by madejejej**
Fixes #307

### CODE REVIEW

File: spec/integration/schema/or_spec.rb [hunk 1]
```diff
@@ -18,6 +18,24 @@
...
+    end
+
+    it "returns success for valid input" do
+      expect(schema.(foo: "bar")).to be_success
+      expect(schema.(foo: 321)).to be_success
+      expect(schema.(foo: true)).to be_success
+    end
+
+    it "provides OR error message for invalid input where all sides failed" do
+      expect(schema.(foo: []).errors.to_h).to eql(foo: ["must be a string or must be an integer or must be boolean"])
```
Comment (madejejej) [inline spec/integration/schema/or_spec.rb:35]: The `eql?` method fails without an explicit `to_h` compared with line 17 - https://github.com/dry-rb/dry-schema/pull/387/files#diff-678244486dbad600e6d79a5bb0cf3aa771877a010ddc921f082f7e3d3bc51e60R17

**Comment by solnic** [comment]
@madejejej why did you close it?

**Comment by madejejej** [comment]
@solnic I won't have the time to work on it and I think it's better if there aren't any dead PRs?

**Comment by solnic** [comment]
@madejejej gotcha, I'm gonna get back to this anyway, just not sure when 🤷🏻

**Pull request closed without merging**